dashboard/client/src/components/DataOverviewTable.tsx

```diff
@@ -116,59 +121,146 @@ const DataOverviewTable = ({
-                progress,
-                total,
-                start_time,
-                end_time,
-              }) => (
-                <TableRow key={dataset}>
-                  <TableCell align="center">{dataset}</TableCell>
-                  <TableCell align="center">
-                    <TaskProgressBar
-                      numFinished={progress}
-                      numRunning={
-                        state === "RUNNING" ? total - progress : undefined
-                      }
-                      numCancelled={
-                        state === "FAILED" ? total - progress : undefined
-                      }
-                      total={total}
-                    />
-                  </TableCell>
-                  <TableCell align="center">
-                    <StatusChip type="task" status={state} />
-                  </TableCell>
-                  <TableCell align="center">
-                    {memoryConverter(Number(ray_data_output_bytes.max))}
-                  </TableCell>
-                  <TableCell align="center">
-                    {memoryConverter(Number(ray_data_current_bytes.value))}/
-                    {memoryConverter(Number(ray_data_current_bytes.max))}
-                  </TableCell>
-                  <TableCell align="center">
-                    {memoryConverter(Number(ray_data_spilled_bytes.max))}
-                  </TableCell>
-                  <TableCell align="center">
-                    {formatDateFromTimeMs(start_time * 1000)}
-                  </TableCell>
-                  <TableCell align="center">
-                    {end_time && formatDateFromTimeMs(end_time * 1000)}
-                  </TableCell>
-                </TableRow>
-              ),
-            )}
+            {list.map((dataset, index) => (
+              <DatasetTable
+                datasetMetrics={dataset}
+                isExpanded={expanded[dataset.dataset as keyof {}]}
+                setIsExpanded={(isExpanded: boolean) => {
+                  const copy = { ...expanded, [dataset.dataset]: isExpanded };
+                  setExpanded(copy);
+                }}
+                key={dataset.dataset}
+              />
+            ))}
           </TableBody>
         </Table>
       </div>
     </div>
   );
 };
 
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    icon: {
+      width: 16,
+      height: 16,
+    },
+  }),
+);
+
+const DatasetTable = ({
+  datasetMetrics,
+  isExpanded,
+  setIsExpanded,
+}: {
+  datasetMetrics: DatasetMetrics;
+  isExpanded: boolean;
+  setIsExpanded: CallableFunction;
+}) => {
+  const classes = useStyles();
+  const {
+    dataset,
+    state,
+    progress,
+    total,
+    ray_data_current_bytes,
+    ray_data_output_bytes,
+    ray_data_spilled_bytes,
+    start_time,
+    end_time,
+    operators,
+  } = datasetMetrics;
+  const operatorRows =
+    isExpanded &&
+    operators.map((operator) => (
+      <TableRow key={operator.operator}>
```

Comment:
Can we refactor `TableRow` into it's own component? This component can then can be re-used between an operator row and a dataset row.